### DISCUSSION TITLE
Add zoom on history

### DISCUSSION THREAD

**Discussion by rlippmann** [home-assistant/frontend#19603] [Other feature requests] [answered]
Hi,

It would be nice if you could somehow easily zoom in and out on the history page.  Currently, it seems that you need to manually adjust the start and end times to do it.

**Comment by KNXBroker** [comment]
A Feature under the 10 Top WTH. 
Agreed, feature would be so appreciated. Openhab does have it, and History-Explorer card shows it is possible. 

https://community.home-assistant.io/t/wth-cant-i-zoom-in-or-out-on-history-graphs/467734/16

**Comment by wrfz** [comment]
I can only agree with that.
The data evaluation and thus the history graph is one of the most important functions of HA. Unfortunately, it is only partially intuitive and relatively inconvenient to use. It is extremely time-consuming if you want to keep moving left or right in time and zooming the data.

The following would significantly improve user-friendliness:
- Hitstory grahph should zoom in and out via mouse wheel or two finger zoom gesture
- It should be possible to swipe horizontally to the left and right over time using a mouse or touch

**Comment by marcinbauer-ohf** [comment, marked as answer]
This has been discussed, and we'll be pursuing this in the near future.

**Reply by wrfz** [reply]
Thanks for the info. Glad to hear it.

One more comment:
It would also be great if you could somehow influence the height of the chart / canvas (have tried different approaches, unfortunately nothing helped), ideally dynamically (stretch).
There are sometimes use-cases where the 400px height of the canvas (on my desktop PC) is too small and you have big problems to recognize details.
